File: study-party-fe/src/services/auth.service.ts
import http from "@/lib/http.ts";
import type { AuthResponse, LoginPayload, RegisterPayload } from "@/types/auth.type.ts";
import { clearTokens } from "@/lib/token.ts";
import type {TokenPair} from "@/types/token.type.ts";

export const login = (payload: LoginPayload) => {
	return http.post<AuthResponse>("auth/login", payload, { withCredentials: true });
};

export const register = (payload: RegisterPayload) => {
	return http.post<void>("auth/register", payload,  { withCredentials: true });
};

export const logout = () => {
	clearTokens();
	return http.post("auth/logout");
};

export const loadMe = () => {
	return http.get("user/me");
};

export const refreshToken = () => {
    return http.post<TokenPair>("auth/refresh", {}, { withCredentials: true });
}

File: party/src/main/java/com/web/study/party/services/auth/AuthService.java
package com.web.study.party.services.auth;

import com.web.study.party.dto.TokenPair;
import com.web.study.party.dto.request.user.*;
import com.web.study.party.dto.response.TokenResponse;
import com.web.study.party.dto.response.auth.AuthResponse;

public interface AuthService {
    void register(RegisterRequest req);
    AuthResponse login(LoginRequest req, String ip, String ua);
    TokenResponse googleLogin(String idToken);
    TokenPair refresh(String refreshToken, String ip, String ua);
    void logout(String refreshToken);
}
